Samples/TicTacToe/OrleansXO.WorkerRole/WorkerRole.cs

```diff
@@ -29,15 +29,28 @@ public override bool OnStart()
             // Set the maximum number of concurrent connections 
             ServicePointManager.DefaultConnectionLimit = 12;
 
+            // Do other silo initialization – for example: Azure diagnostics, etc
+            
             // For information on handling configuration changes
             // see the MSDN topic at http://go.microsoft.com/fwlink/?LinkId=166357.
-
-            silo = new AzureSilo();
-
-            return silo.Start(RoleEnvironment.DeploymentId, RoleEnvironment.CurrentRoleInstance); 
+            return base.OnStart();
         }
 
-        public override void OnStop() { silo.Stop(); }
-        public override void Run() { silo.Run(); } 
+        public override void OnStop() { 
+            silo.Stop();
+            base.Stop();
+        }
+        
+        public override void Run() { 
+            var config = new ClusterConfiguration();
+            config.StandardLoad();
+            
+            // It is IMPORTANT to start the silo not in OnStart but in Run. 
+            // Azure may not have the firewalls open yet (on the remote silos) at the OnStart phase.
+            silo = new AzureSilo();
+            bool isSiloStarted = silo.Start(config);
```

Comment:
Do you mean the deploymentid argument? Because it is optional parameter.

The bug originally was the instantiation and start of the silo in the onstart method, when the firewall of azure was not allowing the connection with table storage 




> On Dec 5, 2015, at 2:12 AM, Jorgen Thelin <notifications@github.com> wrote:
> 
> In Samples/TicTacToe/OrleansXO.WorkerRole/WorkerRole.cs:
> 
> >          }
> >  
> > -        public override void OnStop() { silo.Stop(); }
> > -        public override void Run() { silo.Run(); } 
> > +        public override void OnStop() { 
> > +            silo.Stop();
> > +            base.Stop();
> > +        }
> > +        
> > +        public override void Run() { 
> > +            var config = new ClusterConfiguration();
> > +            config.StandardLoad();
> > +            
> > +            // It is IMPORTANT to start the silo not in OnStart but in Run. 
> > +            // Azure may not have the firewalls open yet (on the remote silos) at the OnStart phase.
> > +            silo = new AzureSilo();
> > +            bool isSiloStarted = silo.Start(config);
> This seems to be missing some way to pass the Azure runtime DeploymentId into the Start method?
> 
> —
> Reply to this email directly or view it on GitHub.
>